main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -320,6 +300,35 @@ object Type {
     */
   def freshEffectVar()(implicit flix: Flix): Type.Var = Type.Var(flix.genSym.freshId(), Kind.Effect)
 
+  /**
+    * Constructs a RecordExtend type.
+    */
+  def mkRecordExtend(label: String, tpe: Type, rest: Type): Type = {
+    mkApply(Type.Cst(TypeConstructor.RecordExtend(label)), List(tpe, rest))
+  }
+
+  /**
+    * Constructs a SchemaExtend type.
+    */
+  def mkSchemaExtend(sym: String, tpe: Type, rest: Type): Type = {
+    mkApply(Type.Cst(TypeConstructor.SchemaExtend(sym)), List(tpe, rest))
+  }
+
+  /**
+    * Constructs a Relation type.
+    */
+  def mkRelation(tuple: Type): Type = {
+    Type.Apply(Type.Cst(TypeConstructor.Relation), tuple)
+  }
+
+  /**
+    * Constructs a Lattice type.
+    */
+  def mkLattice(tuple: Type): Type = {
+    Type.Apply(Type.Cst(TypeConstructor.Lattice), tuple)
+  }
+
```

Comment:
In the above you use TypeConstructor.Relation and TypeConstructor.SchemaExtend, but what about TypeConstructor.Predicate?

Reply:
I don't quite understand what you mean.

`TypeConstructor.Predicate` doesn't exist since we distinguish between lattices and relations.

Lattices and Relations share the kind `Predicate` since they both are used as parameters for Schema construction.

Reply:
Is the idea that schema types take something of kind Predicate and Relation/Lattice are two such things? Or is the idea that you apply a Relation/Lattice to a Predicate?

Reply:
Either way, I think it makes more sense than it did yesterday :)